src/repr/src/row.rs

```diff
@@ -1982,6 +2191,15 @@ mod tests {
                 ))
                 .unwrap(),
             ),
+            Datum::Timestamp(
```

Comment:
This function should also call `ScalarType::enumerate()` then `interesting_datums()` on all things returned there. Might be good to even remove all these datums, and only use that list. If there's things here we want to test, they should be added to interesting_datums.